Assets/Scripts/Spawner/NPCSpawner.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Constants;
+using System;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.AI;
+using Utils;
+
+namespace Spawner
+{
+    public class NPCSpawner : MonoBehaviour
+    {
+        [Tooltip("The area NPCs should be spawned in")]
+        [SerializeField]
+        private Area spawnArea;
+
+        [Tooltip("The maximum NPCs that can be spawned in the area concurrently")]
+        [SerializeField]
+        private int maximumConcurrentEnemies;
+
+        [Tooltip("The probability to spawn an NPC for each frame")]
+        [SerializeField]
+        [Range(0, 0.1f)]
+        private float spawnProbability;
+
+        [Tooltip("The NPC to spawn")]
+        [SerializeField]
+        private GameObject toSpawn;
```

Comment:
You can use the exact type here as well. E.g. EnemyController. With this you can access for example the mesh renderer of it and don't need to use the _**forbidden**_ GetComponentChild :P

Reply:
what do you mean? In toSpawn I put the prefab that should be spawned.. how should I put EnemyController here and how should this help with GetComponent in child?

Reply:
I know GetComponentInChildren is ugly but I couldn't think of a better solution 😔 The alternative is to do it with the RenderChecker, but since this is a lot of stuff, I preferred the other way

Reply:
Instantiate is a generic method:
`public T Instantiate<T>()`
which means it returns the type you give it. You should be able to define it as EnemyController and drag the prefab in. Then, when instantiating, you automatically get the type back and you could call a reference from it e.g. `enemy.Renderer`

Reply:
Very good idea!